src/server/pfs/server/api_server.go

```diff
@@ -648,3 +524,186 @@ func (a *apiServer) RenewFileset(ctx context.Context, req *pfs.RenewFilesetReque
 	}
 	return &types.Empty{}, nil
 }
+
+func modifyFile(ctx context.Context, server modifyFileSource, uw *fileset.UnorderedWriter) (int64, error) {
+	var total int64
+	for {
+		req, err := server.Recv()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				return total, nil
+			}
+			return -1, err
```

Comment:
I think this should still return the bytes read, also I prefer `bytesRead` over `total`.